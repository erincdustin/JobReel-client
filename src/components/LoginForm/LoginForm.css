.LoginForm {
  grid-column-start: 1;
  grid-column-end: 9;
  border: 1px black solid;
  min-width: 300px;
  margin: 0 auto;
  padding: 5%;
  line-height: 150%;
  border-radius: 10px;
  text-align: center;
  margin: auto;
  background: #3a3846;
  color: white;
  box-shadow:  0.5em 0.5em 0.2em rgb(48, 39, 39);
}


@media screen and (max-width: 480px) {
  .LoginForm {
    grid-column-start: 1;
    grid-column-end: 13; 
  }

}