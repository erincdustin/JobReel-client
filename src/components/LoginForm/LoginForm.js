import React, { Component } from 'react'
import { Input, Label } from '../../components/Form/Form'
import AuthApiService from '../../services/auth-api-service'
import JobReelContext from '../../context/JobReelContext';
import Button from '../../components/Button/Button'
import './LoginForm.css'

class LoginForm extends Component {
  static defaultProps = {
    onLoginSuccess: () => {}
  }

  static contextType = JobReelContext;

  state = { error: null }

  firstInput = React.createRef()

  handleSubmit = (e) => {
    e.preventDefault();
    const { username, password } = e.target
    this.setState({ error: null });
      AuthApiService.postLogin({
        username: username.value,
        password: password.value,
      })
        .then(res => {
          username.value = ''
          password.value = ''
          this.context.processLogin(res.authToken);
          this.props.onLoginSuccess();
        })
        .catch(res => {
          this.setState({ error: res.error })
        })
  }

  componentDidMount() {
    this.firstInput.current.focus()
  }

  render() {
    const { error } = this.state;
    return (
        <form className="LoginForm"
          onSubmit={this.handleSubmit}
        >
          <div role='alert' className='error-message'>
            {error && <p>{error}</p>}
          </div>
          <div>
            <Label htmlFor='login-username-input'>
              Username
            </Label>
          <br/>
            <Input
              ref={this.firstInput}
              id='login-username-input'
              name='username'
              required
            />
          </div>
          <div>
            <Label htmlFor='login-password-input'>
              Password
            </Label>
          <br/>
            <Input
              id='login-password-input'
              name='password'
              type='password'
              required
            />
        </div>
        <br/>
          <div>
            <Button type="submit">
              Log In
            </Button>
          </div>
        </form>
    )
  }
}

export default LoginForm;