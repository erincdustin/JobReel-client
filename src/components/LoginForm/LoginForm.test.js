import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LoginForm from './LoginForm'

describe('Login Form smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<LoginForm />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

let focusOnTargetSpy = jest.fn()

jest
  .spyOn(LoginForm.prototype, 'componentDidMount')
  .mockImplementation(focusOnTargetSpy)


describe('Login Form snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(<LoginForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})