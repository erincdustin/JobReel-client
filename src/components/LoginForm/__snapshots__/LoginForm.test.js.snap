// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form snapshot test renders the UI as expected 1`] = `
<form
  className="LoginForm"
  onSubmit={[Function]}
>
  <div
    className="error-message"
    role="alert"
  />
  <div>
    <Label
      htmlFor="login-username-input"
    >
      Username
    </Label>
    <br />
    <ForwardRef
      id="login-username-input"
      name="username"
      required={true}
    />
  </div>
  <div>
    <Label
      htmlFor="login-password-input"
    >
      Password
    </Label>
    <br />
    <ForwardRef
      id="login-password-input"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <br />
  <div>
    <ForwardRef
      type="submit"
    >
      Log In
    </ForwardRef>
  </div>
</form>
`;
